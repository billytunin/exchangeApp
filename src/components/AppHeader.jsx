import React, { Component } from 'react';
import logo from '../assets/logo.svg';

class AppHeader extends Component {
  render() {
    return (
      <header className="app-header">
        <img src={logo} className="app-logo" alt="logo" />
        <h1 className="app-title">Bienvenido a Exchange App</h1>
      </header>
    );
  }
}

export default AppHeader;
