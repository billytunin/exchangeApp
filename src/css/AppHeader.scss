@import 'global-variables.scss';

.app-header {
  background-color: $background-color-one;
  height: 150px;
  padding: 20px;
  padding-top: calc(#{$balance-header-height} + 40px);
  color: white;
  .app-title {
    font-size: 1.5em;
  }
  .app-logo {
    transform: rotate(-11deg);
    height: 80px;
  }
}