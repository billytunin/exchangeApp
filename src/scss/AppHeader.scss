.app-header {
  background-color: $background-color-one;
  height: 150px;
  padding: 20px;
  color: white;
  .app-title {
    font-size: 1.5em;
  }
  .app-logo {
    transform: rotate(-11deg);
    height: 80px;
  }
}