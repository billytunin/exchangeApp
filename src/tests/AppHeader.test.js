import React from 'react';
import AppHeader from '../components/AppHeader.jsx';
import renderer from 'react-test-renderer';

describe('Suite de tests del componente AppHeader', () => {
  it('renderiza el componente sin errores', () => {
    const component = renderer.create(<AppHeader></AppHeader>);
    const componentInstance = component.getInstance();

    expect(componentInstance).toBeDefined();
  });

  it('matchea el ultimo snapshot de AppHeader', () => {
    const component = renderer.create(<AppHeader></AppHeader>);
    const JSONComponent = component.toJSON();

    expect(JSONComponent).toMatchSnapshot();
  });
})
