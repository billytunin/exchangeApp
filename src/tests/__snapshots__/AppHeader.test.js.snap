// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite de tests del componente AppHeader matchea el ultimo snapshot de AppHeader 1`] = `
<header
  className="app-header"
>
  <img
    alt="logo"
    className="app-logo"
    src="logo.svg"
  />
  <h1
    className="app-title"
  >
    Bienvenido a Exchange App
  </h1>
</header>
`;
